src/tests/ttls/enc.cpp

```diff
@@ -80,6 +170,79 @@ void test_ecall()
     t1.join();
 }
 
+void _test_server()
+{
+    constexpr std::string_view kResponse = "HTTP/1.0 200 OK\r\n\r\nBody\r\n";
+    const std::string ttls_cfg =
+        "{\"tls\":{\"Incoming\":{\"*:9010\":{\"cacrt\": \"" + kCACrt +
+        "\", \"clicert\":\"" + kServerCert + "\", \"clikey\":\"" + kServerKey +
+        "\"}}, \"Outgoing\": {\"localhost\": {\"cacrt\": \"\", \"clicert\": "
+        "\"\", \"clikey\": \"\"}}}}";
+    OE_TEST(oe_load_module_host_socket_interface() == OE_OK);
+
+    const int fd = socket(AF_INET, SOCK_STREAM, 0);
+    OE_TEST(fd != -1);
+
+    ert_init_ttls(ttls_cfg.data());
+
+    sockaddr_in sock_addr_in{};
+    sock_addr_in.sin_family = AF_INET;
+    sock_addr_in.sin_port = htons(9010);
+    inet_aton("127.0.0.1", &sock_addr_in.sin_addr);
+    sockaddr sock_addr = reinterpret_cast<sockaddr&>(sock_addr_in);
+
+    OE_TEST(bind(fd, &sock_addr, sizeof(sockaddr)) == 0);
+    OE_TEST(listen(fd, 10) == 0);
+
+    auto t1 = edgeless::ttls::StartTestClient(true);
+
+    sockaddr client_sock{};
+    socklen_t len = sizeof(sockaddr);
+    int client_fd = -1;
+    do
+    {
+        client_fd = accept4(fd, &client_sock, &len, 0);
+    } while (client_fd == -1 && errno == EAGAIN);
+
+    std::string buf(4096, ' ');
+    int ret = -1;
+    do
+    {
+        try
+        {
+            ret = read(client_fd, buf.data(), buf.size());
+        }
+        catch (const std::exception& e)
+        {
+        }
+    } while (ret == -1 && errno == EAGAIN);
+    OE_TEST(ret > 0);
+    OE_TEST(buf.substr(0, 3) == "GET");
+
+    do
+    {
+        try
+        {
+            ret = write(client_fd, kResponse.data(), kResponse.size());
+        }
+        catch (const std::exception& e)
+        {
+        }
```

Comment:
```suggestion
        ret = read(client_fd, buf.data(), buf.size());
    } while (ret == -1 && errno == EAGAIN);
    OE_TEST(ret > 0);
    OE_TEST(buf.substr(0, 3) == "GET");

    do
    {
        ret = write(client_fd, kResponse.data(), kResponse.size());
```
C functions can't throw. And if they can because of our hooks, this is a bug.